src/TelephoneExtractor.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Hedii\Extractors;
+
+use Symfony\Component\DomCrawler\Crawler as DomCrawler;
+
+class TelephoneExtractor extends Extractor
+{
+    /**
+     * An array of found telephones.
+     *
+     * @var array
+     */
+    protected $telephones;
+
+    /**
+     * Extract the telephones contained in the body of the provided dom.
+     *
+     * @param mixed $dom
+     * @param string $url  Will not be used.
+     * @return array
+     */
+    public function extract($dom, $url)
+    {
+        $this->resetTelephones();
+
+
+        $crawler = new DomCrawler($dom, $url);
+        $text = $crawler->filter('body')->text();
+
+        // TODO: make so that it doesnt return empty strings initially
+        preg_match_all("/(?:(?:\\(?(?:00|\\+)([1-4]\\d\\d|[1-9]\\d?)\\)?)?[\\-\\.\\ \\\\\\/]?)?((?:\\(?\\d{1,}\\)?[\\-\\.\\ \\\\\\/]?){0,})(?:[\\-\\.\\ \\\\\\/]?(?:#|ext\\.?|extension|x)[\\-\\.\\ \\\\\\/]?(\\d+))?/mi", $text, $matches, PREG_SET_ORDER);
+
+		$telephone_delimiters = array(' ', '+', '(', ')', '-', '_', '/', '\\');
+
+		if ($matches) {
+			foreach ($matches as $key => $match) {
+				$original = trim(isset($match[0])?$match[0]:'');
+				$country = str_replace($telephone_delimiters, '', trim(isset($match[1])?$match[1]:''));
+				$number = str_replace($telephone_delimiters, '', trim(isset($match[2])?$match[2]:''));
+				$ext = str_replace($telephone_delimiters, '', trim(isset($match[3])?$match[3]:''));
+				$full_number = ( $country !== '' ? '+'.$country.' ' : '' ). $number . ( $ext !== '' ? ', '.$ext : '' );
+				if ( $number !== '' && trim($full_number) !== '' && strlen($number) > 7 ) {
+					$this->telephones[] = $full_number;
```

Comment:
when i test on random web pages, it returns value ending with a dot (eg: '1234567.'). That is obviously not a phone number. You can add this one to the conditions:
`!  $this->endsWith(trim($full_number), '.')`